internal/backend/sessions/sessionworkflows/workflow.go

```diff
@@ -298,12 +298,18 @@ func (w *sessionWorkflow) createEventSubscription(wctx workflow.Context, filter
 		return uuid.Nil, sdkerrors.NewInvalidArgumentError("invalid filter: %w", err)
 	}
 
-	// generate a unique signal id.
 	var signalID uuid.UUID
-	if err := workflow.SideEffect(wctx, func(wctx workflow.Context) any {
-		return uuid.New()
-	}).Get(&signalID); err != nil {
-		return uuid.Nil, fmt.Errorf("generate signal ID: %w", err)
+
+	if did.IsSessionID() {
```

Comment:
did is really hard to reason about and understand without looking for the definition in the code

```suggestion
	if destinationID.IsSessionID() {
```